### PR TITLE
add migration guide for sdk 0.14

### PR DESCRIPTION
**Pull request by camilamacedo86**
add migration guide for sdk 0.14

### CODE REVIEW

File: doc/migration/version-upgrade-guide.md [hunk 1]
```diff
@@ -637,6 +637,77 @@ Replace `*` per verbs in order to solve the issue [671](https://github.com/opera
...
+
+**(Valid for Ansible based-operators only)** 
+- Extract custom Ansible module `k8s_status`, which is now provided by the `operator_sdk.util` Ansible collection. See [developer_guide](https://github.com/operator-framework/operator-sdk/blob/master/doc/ansible/dev/developer_guide.md#custom-resource-status-management) for new usage.
+- Upgrade minimal Ansible version in the init projects from `2.6` to `2.9` for collections support. 
+
+**(Valid for Helm based-operators only)** 
+- Upgraded Helm operator packages and base image from Helm v2 to Helm v3. Cluster state for pre-existing CRs using Helm v2-based operators will be automatically migrated to Helm v3's new release storage format, and existing releases may be upgraded due to changes in Helm v3's label injection
+
+For further detailed information see [CHANGELOG](https://github.com/operator-framework/operator-sdk/blob/master/CHANGELOG.md#v0141)
+
```
Comment (joelanford) [inline doc/migration/version-upgrade-guide.md]: Add another paragraph:

> If you are using any external helm v2 tooling with the your helm operator-managed releases, you will need to upgrade to the equivalent helm v3 tooling.

File: doc/migration/version-upgrade-guide.md [hunk 2]
```diff
@@ -570,8 +570,6 @@ func printVersion() {
 
 ### Go version
```
Comment (joelanford) [inline doc/migration/version-upgrade-guide.md]: Remove the "Go version" header. Ditto for other v0.14.x.

File: doc/migration/version-upgrade-guide.md [hunk 3]
```diff
@@ -637,6 +635,111 @@ Replace `*` per verbs in order to solve the issue [671](https://github.com/opera
...
+	k8s.io/sample-apiserver => k8s.io/sample-apiserver v0.0.0-20191016112829-06bb3c9d77c9
+)
+replace github.com/docker/docker => github.com/moby/moby v0.7.3-0.20190826074503-38ab9da00309 // Required by Helm
+```
+
+- Run `go mod tidy` to update the project modules
+- Run the command `operator-sdk generate k8s` to ensure that your resources will be updated
+- Run the command `operator-sdk generate crds` to regenerate CRDs
+
+### Deprecations on SDK
```
Comment (joelanford) [inline doc/migration/version-upgrade-guide.md]:
```suggestion
### Deprecations
```

File: doc/migration/version-upgrade-guide.md [hunk 4]
```diff
@@ -637,6 +635,111 @@ Replace `*` per verbs in order to solve the issue [671](https://github.com/opera
...
+  # if you add dependencies to this list.`
+  # if you add dependencies to this list.
+collections:
+- operator_sdk.util
+...
+```
+
+**(Valid for Helm based-operators only)** 
+
+The Helm operator packages as its base image was upgrade from Helm v2 to Helm v3. Note that cluster state for pre-existing CRs using Helm v2-based operators will be automatically migrated to Helm v3's new release storage format, and existing releases may be upgraded due to changes in Helm v3's label injection.
```
Comment (joelanford) [inline doc/migration/version-upgrade-guide.md]:
```suggestion
The Helm operator packages and base image were upgraded from Helm v2 to Helm v3. Note that cluster state for pre-existing CRs using Helm v2-based operators will be automatically migrated to Helm v3's new release storage format, and existing releases may be upgraded due to changes in Helm v3's label injection.
```

File: doc/migration/version-upgrade-guide.md [hunk 5]
```diff
@@ -637,6 +635,111 @@ Replace `*` per verbs in order to solve the issue [671](https://github.com/opera
...
+
+With:
+
+```
+sigs.k8s.io/controller-runtime/pkg/client/apiutil.DynamicRESTMapper
+```
+
+### Notable Changes
+
+**(Valid for Ansible based-operators only)**
```
Comment (joelanford) [inline doc/migration/version-upgrade-guide.md]: Looks like you're mixing bold text and headers. This needs to be consistent and have the proper grouping (i.e. Ansible-related notes under an Ansible header and Helm-related notes under a Helm header)

For example

```md
### Notable Changes

#### Ansible Operator

... Ansible stuff here ...

#### Helm Operator

... Helm stuff here ...
```

File: doc/migration/version-upgrade-guide.md [hunk 6]
```diff
@@ -637,6 +635,111 @@ Replace `*` per verbs in order to solve the issue [671](https://github.com/opera
...
+
+```
+sigs.k8s.io/controller-runtime/pkg/client/apiutil.DynamicRESTMapper
+```
+
+### Notable Changes
+
+**(Valid for Ansible based-operators only)** 
+
+Note that the Ansible version in the init projects from `2.6` to `2.9` for collections support. Also, the Ansible module `k8s_status` was extracted and it is now providing the `operator_sdk.util` Ansible collection. See [developer_guide](https://github.com/operator-framework/operator-sdk/blob/master/doc/ansible/dev/developer_guide.md#custom-resource-status-management) for new usage.
```
Comment (joelanford) [inline doc/migration/version-upgrade-guide.md]:
```suggestion
Note that the Ansible version in the init projects was upgraded from `2.6` to `2.9` for collections support. Also, the Ansible module `k8s_status` was extracted and is now provided by the `operator_sdk.util` Ansible collection. See [developer_guide](https://github.com/operator-framework/operator-sdk/blob/master/doc/ansible/dev/developer_guide.md#custom-resource-status-management) for new usage.
```

Comment (joelanford) [reply doc/migration/version-upgrade-guide.md]: > Also, the Ansible module `k8s_status` was extracted and is now provided by the `operator_sdk.util` Ansible collection.

This is a breaking change and should be called out separately.

File: doc/migration/version-upgrade-guide.md [hunk 7]
```diff
@@ -637,6 +635,111 @@ Replace `*` per verbs in order to solve the issue [671](https://github.com/opera
...
+
+#### (Optional) Update the main.yaml file
```
Comment (joelanford) [inline doc/migration/version-upgrade-guide.md]: Given the above breaking change, is this actually optional?

File: doc/migration/version-upgrade-guide.md [hunk 8]
```diff
@@ -637,6 +635,111 @@ Replace `*` per verbs in order to solve the issue [671](https://github.com/opera
...
+...
+```
+
+**(Valid for Helm based-operators only)** 
+
+The Helm operator packages as its base image was upgrade from Helm v2 to Helm v3. Note that cluster state for pre-existing CRs using Helm v2-based operators will be automatically migrated to Helm v3's new release storage format, and existing releases may be upgraded due to changes in Helm v3's label injection.
+
+If you are using any external helm v2 tooling with the your helm operator-managed releases, you will need to upgrade to the equivalent helm v3 tooling.
+
+For further detailed information see [CHANGELOG](https://github.com/operator-framework/operator-sdk/blob/master/CHANGELOG.md#v0141)
```
Comment (joelanford) [inline doc/migration/version-upgrade-guide.md]: The way this is currently organized, this sentence appears to apply only to the Helm operator, when it is really applicable to the entire version's changeset.

File: doc/migration/version-upgrade-guide.md (hunk 1 above)
Comment (camilamacedo86) [reply doc/migration/version-upgrade-guide.md]: Done solved.

File: doc/migration/version-upgrade-guide.md [hunk 9]
```diff
@@ -637,6 +633,108 @@ Replace `*` per verbs in order to solve the issue [671](https://github.com/opera
...
+github.com/operator-framework/operator-sdk/pkg/restmapper.DynamicRESTMapper
+```
+
+With:
+
+```
+sigs.k8s.io/controller-runtime/pkg/client/apiutil.DynamicRESTMapper
+```
+
+### Break Changes
```
Comment (joelanford) [inline doc/migration/version-upgrade-guide.md]:
```suggestion
### Breaking Changes
```

File: doc/migration/version-upgrade-guide.md [hunk 10]
```diff
@@ -637,6 +633,108 @@ Replace `*` per verbs in order to solve the issue [671](https://github.com/opera
...
+```yaml
+...
+dependencies: []
+  # List your role dependencies here, one per line. Be sure to remove the '[]' above,
+  # if you add dependencies to this list.`
+  # if you add dependencies to this list.
+collections:
+- operator_sdk.util
+...
+```
```
Comment (joelanford) [inline doc/migration/version-upgrade-guide.md]:
`````suggestion
The Ansible module `k8s_status` was extracted and is now provided by the `operator_sdk.util` Ansible collection. See [developer_guide](https://github.com/operator-framework/operator-sdk/blob/master/doc/ansible/dev/developer_guide.md#custom-resource-status-management) for new usage.

To use the collection in a role, declare it at the root level in `meta/main.yaml`:

```
collections:
- operator_sdk.util
```

To use it in a playbook, declare it in the play:
```
- hosts: all
  collections:
   - operator_sdk.util
  tasks:
   - k8s_status:
       api_version: app.example.com/v1
       kind: Foo
       name: "{{ meta.name }}"
       namespace: "{{ meta.namespace }}"
       status:
         foo: bar
```

You can also use the fully-qualified name without declaring the collection:
```
   - operator_sdk.util.k8s_status:
       api_version: app.example.com/v1
       kind: Foo
       name: "{{ meta.name }}"
       namespace: "{{ meta.namespace }}"
       status:
         foo: bar
```
`````

File: doc/migration/version-upgrade-guide.md [hunk 11]
```diff
@@ -637,6 +633,108 @@ Replace `*` per verbs in order to solve the issue [671](https://github.com/opera
...
+
+### Notable Changes
+
+These notable changes contain just the most important user-facing changes. See the [CHANGELOG](https://github.com/operator-framework/operator-sdk/blob/master/CHANGELOG.md#v0141) for details of the release.
+
+#### Ansible version update
+
+Note that the Ansible version in the init projects from `2.6` to `2.9` for collections support. Update the `main.yaml` file :
```
Comment (joelanford) [inline doc/migration/version-upgrade-guide.md]:
```suggestion
The Ansible version in the init projects was upgraded from `2.6` to `2.9` for collections support. Update the `meta/main.yaml` file.
```

File: doc/migration/version-upgrade-guide.md [hunk 12]
```diff
@@ -637,6 +633,108 @@ Replace `*` per verbs in order to solve the issue [671](https://github.com/opera
...
+replace github.com/docker/docker => github.com/moby/moby v0.7.3-0.20190826074503-38ab9da00309 // Required by Helm
+```
+
+- Run `go mod tidy` to update the project modules
+- Run the command `operator-sdk generate k8s` to ensure that your resources will be updated
+- Run the command `operator-sdk generate crds` to regenerate CRDs
+
+### Deprecations
+
+Note that `github.com/operator-framework/operator-sdk/pkg/restmapper` was deprecated in favor of the `DynamicRESTMapper` implementation in [controller-runtime](https://godoc.org/github.com/kubernetes-sigs/controller-runtime/pkg/client/apiutil#NewDiscoveryRESTMapper). Users should migrate to controller-runtime's implementation, which is a drop-in replacement.
```
Comment (joelanford) [inline doc/migration/version-upgrade-guide.md]:
```suggestion
The `github.com/operator-framework/operator-sdk/pkg/restmapper` package was deprecated in favor of the `DynamicRESTMapper` implementation in [controller-runtime](https://godoc.org/github.com/kubernetes-sigs/controller-runtime/pkg/client/apiutil#NewDiscoveryRESTMapper). Users should migrate to controller-runtime's implementation, which is a drop-in replacement.
```

File: doc/migration/version-upgrade-guide.md [hunk 13]
```diff
@@ -637,6 +633,130 @@ Replace `*` per verbs in order to solve the issue [671](https://github.com/opera
...
+
+```
+sigs.k8s.io/controller-runtime/pkg/client/apiutil.DynamicRESTMapper
+```
+
+### Breaking Changes
+
+#### Add `operator_sdk.util` Ansible collection
+
+Also, the Ansible module `k8s_status` was extracted and it is now providing the `operator_sdk.util` Ansible collection. See [developer_guide](https://github.com/operator-framework/operator-sdk/blob/master/doc/ansible/dev/developer_guide.md#custom-resource-status-management) for new usage. Update the end of the `main.yaml` file with the `collections` after the `dependencies` block:
```
Comment (joelanford) [inline doc/migration/version-upgrade-guide.md]: Remove the above line.

Comment (camilamacedo86) [inline doc/migration/version-upgrade-guide.md]:
```suggestion
```

**Pull request merged**